[9fc70c892] iP

9fc70c892 opened the pull request:
# Waffles
> The best things in life are waffles. 

🧇 An *alternative* to Duke. 🧇

Are you looking for task manager that is:
1. **Distraction Free**
2. Text-based
3. Portable
4. Open Sourced

Waffles got you covered.

### Features:
- [x] Keeps tracks of your tasks
- [ ] Reminders (coming soon)

### Download:
* [Download](https://github.com/9fc70c892/cs2030s-ay2021)


---
If you are a Java programmer, you can experiment or add on to our [Source Code](https://github.com/9fc70c892/cs2030s-ay2021)!

```Java
public class Main {
    public static void main(String[] args) {
        Duke.main(args);
    }
}
```

9fc70c892 committed:
Level-1
Rename splash to Waffles.
Added Greeting, Echo input, Exit with bye

9fc70c892 committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

9fc70c892 committed:
Fixed logo
Added Add, List for Level 2 (Note : No sanitization and checks if index is possible or not)

9fc70c892 committed:
Minimal accomplishment of Level-4. Look into changing how input is handled and breaking switch cases into functions, potentially creating a 'taskmaster' class.

9fc70c892 committed:
Tentative solution to A-TextUiTesting, to update later post L6

9fc70c892 committed:
Some work on exceptions, real exception class to come later but requires rework of input handling.
Also current input allows missing description provided /by is present i.e. `deadline /by string`.

9fc70c892 committed:
Simple delete function. Again, to rush for completion due to CS2109 pushback. This req overhaul.

9fc70c892 committed:
Upgraded all.

9fc70c892 committed:
Clicked stage all when supposed to not stage this. Deleted.

9fc70c892 committed:
Add Save Load Functionality

Added functionality as required by Level-7.

9fc70c892 committed:
Added Level-8 DateTime Requirements

Minimum handling of datetime has been added. Expect merge conflict with
other branches due to changing child classes of TaskMaster input.

9fc70c892 committed:
Merge branch 'branch-Level-7'

9fc70c892 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/TaskMaster.java

9fc70c892 committed:
Partial Refactoring of Code.

Incomplete, however, this is primary commit.

9fc70c892 committed:
Code Refactoring part 2

Broke code due to parser.

9fc70c892 committed:
Code Refactoring part 3

Fixed errors due to refactor. See issues for improvable items.

9fc70c892 committed:
Add Gradle and Jar support

Completion of Minimal criteria for A-Gradle, less style check

9fc70c892 committed:
Added A-JUnit support

Minimal auto testing of two methods to fulfill A-JUnit

9fc70c892 committed:
Add JavaDoc for some methods

Commit to mark that JavaDocs were added previously in previous weeks.
Yet to standardize and fill all.

9fc70c892 committed:
Add Missing Functionality Level-7

Level-7 Required auto loading upon startup, this was not added in
for that particular tag.

9fc70c892 committed:
Revert "Add Missing Functionality Level-7"

This reverts commit 894f944c17f96667f7f6ca9cf05395661eb9e153.

9fc70c892 committed:
Add Functionality Missing for Level 7

Added auto load functionality that was missing during Level-7 Push

9fc70c892 committed:
Delete File Gradle Test

9fc70c892 committed:
Merge branch 'branch-Level-7'

9fc70c892 committed:
Add Find

Simple find feature added to parser

9fc70c892 committed:
Merge branch 'branch-Level-9'

9fc70c892 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/core/DukeIO.java

9fc70c892 committed:
Add save upon exit

9fc70c892 committed:
Merge branch 'branch-Level-7'

9fc70c892 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/core/DukeIO.java

9fc70c892 committed:
Modify code to satisfy Gradle Checkstyle

9fc70c892 committed:
Add missing Javadoc based on Gradle Codestyle Check

9fc70c892 committed:
Merge branch 'branch-A-JavaDoc'

9fc70c892 committed:
Edit README.md to match PR

9fc70c892 committed:
Change variable and function names

9fc70c892 committed:
Add GUI

Based on fx guide

9fc70c892 committed:
Merge tag 'Level-10'

9fc70c892 committed:
Fix error message on startup

Greetings wrongly fed into getResponse.

jerome-neo commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,54 @@
+     * Returns the name of the Task.
+     * @return The description or name of this Task.
+     */
+    public String getTaskDescription() {
```
Use of appropriate naming conventions for class methods in your task package. Names also follow camelCase, e.g. toCsv() and not toCSV(), which is a common mistake. Names of boolean sound like booleans.👍

9fc70c892 replied:
Thank you for your kind words!

jerome-neo commented on src/main/java/core/Parser.java:
```diff
@@ -0,0 +1,212 @@
+     * @return Message from the corresponding commands to print out.
+     * @throws DukeException Thrown when missing parameters is found or index error etc.
+     */
+    public String parse(TaskMaster tm) throws DukeException {
```
Really neat and easy to read 👍, also followed convention and carefully avoided indentation for case clauses.

9fc70c892 replied:
Thank you for your kind words!

jerome-neo reviewed (commented):
Your code looks really neat and readable. To summarise, the names of methods are in camelCase, abbreviations of acronyms were not uppercased when used as part of a name, booleans of labels sound like boolean, brackets in Egyptian style, correct white spacing and every class is part of some package.

shenchenzizoe commented on src/main/java/core/Parser.java:
```diff
@@ -0,0 +1,212 @@
+        }
+        try {
+            int ind = Integer.parseInt(userInSplit[0]);
+            return ind - 1; // Count starting from 0
```
might be clearer if name the ind variable

9fc70c892 replied:
Thank you for your kind words.

Thank you for pointing out my mistake. It looks like I have missed out this particular part by accident.

shenchenzizoe commented on src/main/java/core/Parser.java:
```diff
@@ -0,0 +1,212 @@
+     * @param tm   The runtime TaskMaster object
+     * @throws DukeException Thrown when input is invalid, which can appear when user manually edits the file.
+     */
+    public static void parseSaveFile(String[] task, TaskMaster tm) throws DukeException {
```
good implementation of Enums!

shenchenzizoe commented on src/main/java/ux/MainWindow.java:
```diff
@@ -0,0 +1,60 @@
+
+    @FXML
+    public void initialize() {
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
```
type here? vvalueProperty might be valueProperty

9fc70c892 replied:
Hi thank you for pointing this out.

I [have checked](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/ScrollPane.html#vvalueProperty--). And [it does not seem to be a mistake](https://se-education.org/guides/tutorials/javaFxPart4.html#using-controllers), but I will double check if it affects the code flow.

shenchenzizoe reviewed (commented):
Clear structure and proper use of packages! a few coding standard issues could be modified tho.

9fc70c892 committed:
Edit Code based on PR comments

Made changes to code based on PR comments provided by other students
to improve code quality

9fc70c892 committed:
Improve Code Readability

9fc70c892 committed:
Add Assertion to Parser

9fc70c892 committed:
Add Reminder

Added B-Reminder. Reads all deadline and gets the one with the
latest deadline. Works with multiple deadlines of the same time.

9fc70c892 committed:
Code style cleanup

9fc70c892 committed:
Merge branch 'branch-B-Reminder'

9fc70c892 committed:
Merge branch 'master' into branch-A-Assertions

9fc70c892 committed:
Merge branch 'branch-A-Assertions' into branch-A-CodeQuality

9fc70c892 committed:
Test for Add Increments as PR

9fc70c892 committed:
Merge pull request #8 from 9fc70c892/branch-A-CodeQuality

Test for Add Increments as PR

9fc70c892 committed:
Add User Guide

9fc70c892 committed:
Update GUI to show all when listing

9fc70c892 committed:
Fix remove and date string format

Fixed remove to fix issue posted on github.
Realized that date formatting is not correct.

9fc70c892 committed:
Cleanup Javadoc and dead commands

9fc70c892 committed:
Add Test and fix readme.md

9fc70c892 committed:
Update Readme link

9fc70c892 committed:
Fix User Guide location

9fc70c892 committed:
Update README.md
